### ISSUE TITLE
Convert Numba to use `pyproject.toml`

### ISSUE THREAD

**Issue by esc** [numba/numba#9060]
## Feature request

Convert Numba to use `pyproject.toml` instead of `setup.py`

**Comment by esc** [comment]
As per https://github.com/numba/numba/wiki/Minutes_2023_11_07 I will remove this from the 0.59.0-rc1 milestone, as it is no longer planned.

**Comment by detrin** [comment]
I really would like to see this in future At this moment setup.py is obsolete and rather archaic

**Comment by esc** [comment]
@detrin I agree, it would be absolutely lovely if someone could find time for this. Sadly, nothing is concretely broken, so there is no immediate need to fix this.

**Comment by esc** [comment]
IIRC I did look into this a while ago. The specific issue I encountered is that `pyproject.toml` doesn't support `versioneer`, only `setuptools_scm`. You can see an example of using `pyproject.toml` with `setuptools_scm` here:

https://github.com/numba/pixie/blob/main/pyproject.toml

The second major hurdle here is that all CI systems and buildscripts and conda recipes and so on will need to be adapted. However the CI setup is currently undergoing major overhaul. 

Since we don't want to have that many moving parts at the same time, I would recommend to park this until the CI overhaul is complete (i.e. couple of months from now at the earliest).

**Comment by detrin** [comment]
Okay good to know. If there is an update ping me. Possibly when I have time over following months. I might look into it myself.
